build-system/pr-check/build-targets.js

```diff
@@ -380,8 +380,22 @@ function buildTargetsInclude(...targets) {
   return Array.from(targets).some((target) => buildTargets.has(target));
 }
 
+/**
+ *
+ * @return {boolean}
+ */
+function requiresBuild() {
+  return buildTargetsInclude(
+    Targets.RUNTIME,
+    Targets.INTEGRATION_TEST,
+    Targets.E2E_TEST,
+    Targets.VISUAL_DIFF
+  );
+}
```

Comment:
This logic isn't universally true because of how different tasks use different build flavors. E.g. Not all of our tests actually use the module build.

I believe @danielrozenberg has a better fix in mind.

Reply:
I'm planning on reverting this change after validating your theory about percy.